### PR TITLE
further documentation refactoring

### PR DESCRIPTION
**Pull request by Zethson**
Time to fix the remaining minor documentation issues.

I've refactored the troubleshooting page out of development section. Maybe it is better to have one in development and one in usage? 
I feel like the production setup instructions are a little bit outdated. I, for one, like to work with Docker anyways.
Please point out missing or wrong documentation related to the production setup and I will fix it.

Signed-off-by: Zethson <lukas.heumos@posteo.net>

### CODE REVIEW

File: docs/source/docs/installation/installation_prod.inc [hunk 1]
```diff
@@ -8,26 +8,21 @@ containerized environment please refer to :ref:`docker_installation`.
...
-   conda install -c bioconda megaqc
-
 2. Export environment variables
 -------------------------------
 
 By default, MegaQC runs in development mode with a sqlite flat file
 database (this is to make it as simple as possible to get up and running
 for a quick test / demo). To tell MegaQC to use a production server, you
-need to set the ``MEGAQC_PRODUCTION`` environment variable to true.
+need to set the ``MEGAQC_PRODUCTION`` environment variable to true (=1).
```
Comment (TMiguelT) [inline docs/source/docs/installation/installation_prod.inc:25]: I would still advise giving an example here. It's not intuitive how to represent a boolean via env vars

File: docs/source/docs/installation/installation_prod.inc [hunk 2]
```diff
@@ -171,4 +178,5 @@ If you run an older OS, ensure that the package is installed.
 
 .. _Python MySQL connector: https://dev.mysql.com/downloads/connector/python/2.1.html
 .. _PyPI package: https://pypi.python.org/pypi/mysql-connector-python/2.0.4
-.. _create an issue: https://github.com/ewels/MegaQC/issues/new
\ No newline at end of file
+.. _create an issue: https://github.com/ewels/MegaQC/issues/new
+.. _deployment_folder: https://github.com/ewels/MegaQC/blob/master/deployment
```
Comment (TMiguelT) [inline docs/source/docs/installation/installation_prod.inc:182]: Add final newline.

File: docs/source/docs/installation/installation_prod.inc [hunk 3]
```diff
@@ -107,9 +100,20 @@ Then in this file, set the following configuration key pair:
 This should, hopefully, make everything work. If you have problems,
 please `create an issue`_ and we’ll do our best to help.
 
-4. (Optional, but recommended) Create an apache proxy
+4. Install the JavaScript and start compiling
```
Comment (TMiguelT) [inline docs/source/docs/installation/installation_prod.inc:103]: The way we've designed it, you shouldn't have to have compile the JS or even node installed for the production build. If you `pip install megaqc` it should contain the pre-compiled JS. If this is not the case then that's a bug, but nonetheless this section shouldn't be needed in the docs. See #106 for context.

**Review by TMiguelT — CHANGES_REQUESTED** [review]
* What's up with the `.inc` file extension. Are they not `.rst` files?
* Didn't we establish that `psycopq2-binary` is not recommended for production?

File: docs/source/docs/installation/installation_dev.inc [hunk 4]
```diff
@@ -16,7 +16,7 @@ If you’re doing development work, you need access to the source code
 
 .. code:: bash
 
-   git clone git@github.com:ewels/MegaQC.git
+   git clone https://github.com/ewels/MegaQC
```
[the remark this answers is not in the document]
Comment (Zethson) [reply docs/source/docs/installation/installation_dev.inc:19]: The https:// clone URLs are available on all repositories, regardless of visibility. https:// clone URLs work even if you are behind a firewall or proxy.

```
zeth@master /tmp> git clone git@github.com:ewels/MegaQC.git                                                                                                                            (base) 
Cloning into 'MegaQC'...
Warning: Permanently added the RSA host key for IP address '140.82.121.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by Zethson** [comment]
@TMiguelT The renaming of the *.rst files, which are included into other *.rst files was to prevent duplicate label warnings.
See: https://stackoverflow.com/questions/16262163/sphinxs-include-directive-and-duplicate-label-warnings

> Didn't we establish that psycopq2-binary is not recommended for production?
Yes. Did I change anything in the docs about that? I can't find it. setup.py also looks fine?

**Comment by TMiguelT** [comment]
Right. I'm not a huge fan of the `.inc` because having the wrong file extension breaks a lot of tooling (e.g. the github previews, diffs, and also my editor settings).

The `psycopg` comment was in relation to this: https://github.com/ewels/MegaQC/blob/82f9f7900649d3cc238cb286b7bec4ec9c59af34/docs/source/docs/installation/installation_prod.inc#L61

**Comment by Zethson** [comment]
@TMiguelT I removed the line that you pointed it.

I am also not a fan of the "wrong" file extension, but it may be the best solution.

We could either
* use *.rst, ignore the warnings and hope for the best
* don't include files but write everything into one big file
* use *.inc and lose editor support

I went for solution 3 and consider it the best of bad solutions.

Do you have other ideas?

**Comment by TMiguelT** [comment]
What are your thoughts on the `include_patterns` suggestion mentioned in the link you posted?

**Comment by Zethson** [comment]
Works, but then we'll have to write every include file into the patterns:

```
exclude_patterns = ["_build",
                    "Thumbs.db",
                    ".DS_Store",
                    "docs/installation/installation_dev.rst",
                    "docs/installation/installation_docker.rst",
                    "docs/installation/installation_prod.rst",
                    "docs/installation/migrations.rst"]
```

If you prefer this solution I can implement it.

**Comment by TMiguelT** [comment]
I do think that's my preference, because we gain a lot of tooling, plus it becomes easier to understand the codebase (I was confused as to what type of file this was). If someone adds a new include and forgets to update the `exclude_patterns` then all that will happen is a warning, right? That seems fine. Just chuck in a comment to the `conf.py` that explains why we are doing that.

**Comment by Zethson** [comment]
It may just be a warning, but it as well break a label if someone forgets it.
But a code review here should raise alarm bells when new files are included and they're labeled in various places.

**Comment by Zethson** [comment]
@TMiguelT done.

**Review by TMiguelT — APPROVED** [review]

**Pull request merged**